Sources/race.swift

```diff
@@ -11,10 +11,14 @@
  - Returns: A new promise that resolves when the first promise in the provided promises resolves.
  - Warning: If any of the provided promises reject, the returned promise is rejected.
 */
-public func race<T>(promises: Promise<T>...) -> Promise<T> {
+public func race<T>(promises: [Promise<T>]) -> Promise<T> {
     return Promise(sealant: { resolve in
```

Comment:
There's an issue here if the array param is empty, the promise will never fulfill. I've handled this in #374 but could really use your tests, would you like to merge?